### ISSUE TITLE
alternative parametrization of Negative Binomial

### ISSUE THREAD

**Issue by fkrauer** [TuringLang/DistributionsAD.jl#218]
Hi everyone

Are there any plans to implement an alternative parametrization of the negative binomial distributions, which takes the mean and the overdispersion parameter as arguments, see [here](https://mc-stan.org/docs/2_20/functions-reference/nbalt.html) (and [here](https://esajournals.onlinelibrary.wiley.com/doi/full/10.1890/10-1831.1))?

I have implemented a helper function to fill this gap (with ψ (0,1]), but this probably slows down things due to type instability issues? Thanks
```
function NegativeBinomial2(ψ, mu)
    p = 1.0/(1.0 + ψ*mu)
    r = 1.0/ψ
    return NegativeBinomial(r, p)
end
```

**Comment by devmotion** [comment]
> Are t[here](https://mc-stan.org/docs/2_20/functions-reference/nbalt.html) any plans to implement an alternative parametrization of the negative binomial distribution, which takes the mean and the overdispersion parameter as arguments, see [here](https://esajournals.onlinelibrary.wiley.com/doi/full/10.1890/10-1831.1) (and here)?

No. I think it would fit better in Distributions, possibly as a separate distribution (similar to e.g. `Normal` and `NormalCanon`). IMO it is out of scope of DistributionsAD, and the long term plan is also to get rid of DistributionsAD and instead move all fixes/workarounds upstream to e.g Distributions and AD packages in order to get rid of the type piracies here and simplify maintenance.

> I have implemented a helper function to fill this gap (with ψ (0,1]), but this probably slows down things due to type instability issues?

I'm not sure which type instability you are referring to, the function seems to be type stable? As a general recommendation, often it is useful to avoid `Float64` literals if the arguments can be of other number types than `Float64`, mainly to avoid undesired promotions. Also often people expect that a function named `NegativeBinomial2` is a constructor and returns objects of type `NegativeBinomial2` - but this is only a convention. Both things (`Float64` literals and the name of the function) don't affect the type stability.
